src/Grids/latitude_longitude_grid.jl

```diff
@@ -722,3 +722,7 @@ end
 
 @inline λspacing(i, j, k, grid::LatLonGrid, ℓx, ℓy, ℓz) = λspacing(i, grid, ℓx)
 @inline φspacing(i, j, k, grid::LatLonGrid, ℓx, ℓy, ℓz) = φspacing(j, grid, ℓy)
+
+isxregular(::LatitudeLongitudeGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Number}) = true
+isyregular(::LatitudeLongitudeGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Number}) = true
+iszregular(::LatitudeLongitudeGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Number}) = true
```

Comment:
Just a clarification about the nomenclature here. `LatLonGrid`s can be regular in longitude (i.e. when we're talking about degrees), but in that case Δx (i.e. in meters) can vary with latitude (even though they don't vary with longitude), no?

Do we still call that _regular_ x-spacing? Or do we somehow need to differentiate things a bit further?

Reply:
Where are these functions used? That should determine the correct return value. Then we can worry about the name.

Reply:
In interpolation it is being assumed that we're talking about meters rather than longditude since we call `xspacings` and `xnodes` etc. which return in meters.

We probably need a way to tell if a user is interpolating in meters of degrees (although I don't think this should be a problem for how interpolation is used internally).

Reply:
Just recently the definition of `xnode` has changed for the `LatitudeLongitudeGrid` from degrees to meters. The particle advection assumes that the `x`, `y` coordinates of the particles are in degrees for a `LatitudeLongitudeGrid`

Reply:
I think it would be most straight forward if the particles were always in meters? Otherwise we need to somehow detect in the interpolaiton if the user wants to interpolate in degrees or meters and I don't think interpolation is used anywhere else where it might be in degrees?

Reply:
Changing particles in this PR as well as #2928 will be painful so it would probably be easiest to wait and change after thats merged

Reply:
Its should be quite easy to change to Lagrangian particles being in meters since we would just need to remove the `corrdinate_transform_u` etc., it might be more challenging to change the `return_face_metrics` function

Reply:
On the other hand, changing to work in meters will make most `LatitudeLongditudeGrid` particle simulaitons much slower because most will have regular lat/lon but not meters so it probably would be better to stick with lat/lon?